subcommand/server-acl-init/command_ent_test.go

```diff
@@ -0,0 +1,436 @@
+// +build enterprise
+
+package serveraclinit
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"testing"
+)
+
+// Test that ACL policies get updated if namespaces config changes.
+func TestRun_ACLPolicyUpdates(t *testing.T) {
+	t.Parallel()
+	k8s, testAgent := completeEnterpriseSetup(t, resourcePrefix)
+	defer testAgent.Stop()
+	require := require.New(t)
+
+	ui := cli.NewMockUi()
+	firstRunArgs := []string{
+		"-server-label-selector=component=server,app=consul,release=" + releaseName,
+		"-resource-prefix=" + resourcePrefix,
+		"-k8s-namespace=" + ns,
```

Comment:
Besides adding an update scenario to going to mirroring, it would also be good to test the case where the k8s namespace is not `default`.

Reply:
Do you mean testing the policy updates from when the kubernetes namespace is not default or another type of update? AFAIK the k8s namespace doesn't affect the policies at all right?

Reply:
So we both agree that this flag has no effect on whether the policies are updated and we've already tested the templating of the policies in the rules_test.go. It sounds like you're saying the reason we should test this flag varying is because it's related to namespaces. I don't think that's a good enough reason to add an additional case here. You could make the same argument for the -allow-k8s-namespace flag. In the end I think that would complicate the test without a corresponding gain in meaningful testing coverage.

Reply:
It's a test against future human error, where someone modifies the wrong namespace flag.

Reply:
Okay I added it.